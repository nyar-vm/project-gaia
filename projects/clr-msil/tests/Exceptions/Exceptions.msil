.class public auto ansi beforefieldinit
  Valkyrie.Translator.Exceptions.Exceptions
    extends [System.Runtime]System.Object
{

  .method public hidebysig static void
    ThrowException(
      string message
    ) cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.NullableContextAttribute::.ctor(unsigned int8)
      = (01 00 01 00 00 ) // .....
      // unsigned int8(1) // 0x01
    .maxstack 8

    // [6 5 - 6 6]
    IL_0000: nop

    // [7 9 - 7 38]
    IL_0001: ldarg.0      // message
    IL_0002: newobj       instance void [System.Runtime]System.Exception::.ctor(string)
    IL_0007: throw

  } // end of method Exceptions::ThrowException

  .method public hidebysig specialname rtspecialname instance void
    .ctor() cil managed
  {
    .maxstack 8

    IL_0000: ldarg.0      // this
    IL_0001: call         instance void [System.Runtime]System.Object::.ctor()
    IL_0006: nop
    IL_0007: ret

  } // end of method Exceptions::.ctor
} // end of class Valkyrie.Translator.Exceptions.Exceptions
