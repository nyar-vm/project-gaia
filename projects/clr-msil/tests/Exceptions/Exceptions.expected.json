{
    "assembly_name": null,
    "extern_assemblies": [],
    "module_name": null,
    "class_name": "Valkyrie.Translator.Exceptions.Exceptions",
    "class_modifiers": [
        "public",
        "auto",
        "ansi",
        "beforefieldinit"
    ],
    "extends": "[System.Runtime]System.Object",
    "method_count": 2,
    "method_names": [
        "ThrowException",
        ".ctor"
    ],
    "statement_count": 1
}